data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/AbstractGetSimpleDataFromState.java

```diff
@@ -39,6 +40,10 @@ public AbstractGetSimpleDataFromState(
   public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     final SafeFuture<Optional<StateAndMetaData>> future =
         chainDataProvider.getBeaconStateAndMetadata(request.getPathParameter(PARAMETER_STATE_ID));
+    if (future == null) {
```

Comment:
It's really awkward that future could wind up being null here. But I don't see how this would ever be null looking at the code - it looks like chainDataProvider would throw `BadRequestException` and that should then be handled but the default exception handlers to respond with bad request. Adding a `.exceptionally` block that checks if the root cause of the exception is `BadRequstException` and returns a response with the custom "check state_id is correct" message would be fine (just rethrow any other exceptions).

Reply:
test was wrong - an exception gets handled outside the handler fn, so this null check has been removed.